src/metrics/metrics.ts

```diff
@@ -0,0 +1,214 @@
+import type {LogSink} from '@rocicorp/logger';
+
+export interface ReporterOptions {
+  datadogApiKey: string;
+  metrics: Metrics;
+  // intervalSecs defaults to 2 minutes.
+  intervalSecs?: number;
+  abortSignal?: AbortSignal;
+  // If a LogSink is not provided, the Reporter is silent.
+  logSink?: LogSink;
+  // Should probably also take tags (eg, env, version). Will
+  // probalby need to take host and service too for the server side.
+}
+
+// Reporter periodically reports metrics to Datadog. It uses an interval
```

Comment:
JSDoc is better because it gets included in autocomplete etc

Reply:
> JSDoc

*puke* ok